core/src/test/java/org/elasticsearch/routing/AliasRoutingIT.java

```diff
@@ -40,13 +48,111 @@
 /**
  *
  */
-public class AliasRoutingTests extends ESIntegTestCase {
+public class AliasRoutingIT extends ESIntegTestCase {
+
+    public static class MockScriptPlugin extends Plugin {
+
+        public MockScriptPlugin() {
+        }
+
+        @Override
+        public String name() {
+            return MockScriptEngine.NAME;
+        }
+
+        @Override
+        public String description() {
+            return "Mock script engine for " + AliasRoutingIT.class;
+        }
+
+        public void onModule(ScriptModule module) {
+            module.addScriptEngine(MockScriptEngine.class);
+        }
+
+    }
+
+    public static class MockScriptEngine implements ScriptEngineService {
```

Comment:
Yeah - I don't care about abstract - just that it'd be nice if you would refactor this so it looks its more obvious what is special about each of the mock engines without having to read 20 lines of boiler plate each.